### PR TITLE
rootless: move join namespace inside child process

### PR DESCRIPTION
**Pull request by giuseppe**
open the namespace file descriptors inside of the child process.

Closes: https://github.com/containers/libpod/issues/5873

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

### CODE REVIEW

**Comment by mheon** [comment]
```In function 'join_namespace_or_die':\nrootless_linux.c:544:30: error: format '%ld' expects argument of type 'long int', but argument 3 has type 'int' [-Werror=format=]\n```

**Comment by mheon** [comment]
LGTM once that is addressed, though. Should we consider backport against 1.9?

File: pkg/rootless/rootless_linux.c [hunk 1]
```diff
@@ -535,8 +535,29 @@ create_pause_process (const char *pause_pid_file_path, char **argv)
     }
 }
 
+static void
+join_namespace_or_die (int pid_to_join, const char *ns_file)
+{
+  char ns_path[32];
+  int fd;
+
+  sprintf (ns_path, "/proc/%ld/ns/%s", pid_to_join, ns_file);
```
[the remark this answers is not in the document]
Comment (giuseppe) [reply pkg/rootless/rootless_linux.c]: done

**Comment by giuseppe** [comment]
> LGTM once that is addressed, though. Should we consider backport against 1.9?

comments addressed.

Yes, I think we should backport it.  I'll prepare it once this one is merged

**Comment by rhatdan** [comment]
Still blocked by gating, but LGTM

**Pull request merged**